src/Sarif.UnitTests/Writers/PrereleaseCompatibilityTransformerTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.UnitTests.Writers
+{
+    public class PrereleaseCompatibilityTransformerTests
+    {
+        [Fact]
+        public void PrereleaseCompatibilityTransformer_UpgradePrereleaseTwoZeroZero()
+        {
+            string comprehensiveSarifPath = Path.Combine(Environment.CurrentDirectory, @"v2\ObsoleteFormats\ComprehensivePrereleaseTwoZeroZero.sarif");
+
+            string sarifText = File.ReadAllText(comprehensiveSarifPath);
+
+            sarifText = PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(sarifText);
+
+            // If this obsolete prerelease v2 file deserializes to the current
+            // version without raising an exception, this step succeeds.
```

Comment:
>succeeds [](start = 63, length = 8)

You can say that explicitly in FluentAssertions without requiring the comment, and with a more informative error message if the test fails:
```
Action action = () => {
    SarifLog sarifLog = JsonConvert.DeserializeObject<SarifLog>(sarifText);
    JsonConvert.SerializeObject(sarifLog);
};

action.Should().NotThrow();
```

Reply:
well, every line of the test is implicitly expected not to throw. I had this comment because the deserialization was previously the last line of the test. Now it isn't. I'll just delete the comment.

---
In reply to: [226748533](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226748533) [](ancestors = 226748533)

Reply:
I won't press this point to the extent of reactivating this comment, but -- there are two advantages to `ShouldNotThrow`:
1. It makes it immediately obvious to the reader of the test what you are asserting.
2. If the assertion fires, you get a nice message like "`JsonConvert.SerializeObject(sarifLog) was not expected to throw an exception, but it threw `ArgumentNullException`" -- again making it clear what expectation was violated.

---
In reply to: [226843996](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226843996) [](ancestors = 226843996,226748533)